src/platform/src/Bridge/OpenAi/DallE/ModelClient.php

```diff
@@ -23,18 +24,24 @@
  *
  * @author Denis Zunke <denis.zunke@gmail.com>
  */
-final readonly class ModelClient implements ModelClientInterface
+final readonly class ModelClient extends AbstractOpenAiModelClient implements ModelClientInterface
 {
+    private string $baseUrl;
+
+    /**
+     * @param AbstractOpenAiModelClient::REGION_EU|AbstractOpenAiModelClient::REGION_US|null $region
+     */
     public function __construct(
         private HttpClientInterface $httpClient,
         #[\SensitiveParameter] private string $apiKey,
+        ?string $region = null,
     ) {
-        if ('' === $apiKey) {
-            throw new InvalidArgumentException('The API key must not be empty.');
-        }
+        parent::validateApiKey($apiKey);
```

Comment:
```suggestion
        self::validateApiKey($apiKey);
```

Which gives the ability to override the logic in this class